src/backend/mi2/mi2.ts

```diff
@@ -556,22 +556,22 @@ export class MI2 extends EventEmitter implements IBackend {
         x: 'hexadecimal'
     };
 
-    public async varCreate(expression: string, name: string = '-', scope: string = '@'): Promise<VariableObject> {
+    public async varCreate(expression: string, name: string = '-', fmt: string = null, scope: string = '@'): Promise<VariableObject> {
         if (trace) {
             this.log('stderr', 'varCreate');
         }
-        let fmt = null;
         expression = expression.trim();
         if (/,[bdhonx]$/i.test(expression)) {
             fmt = expression.substring(expression.length - 1).toLocaleLowerCase();
             expression = expression.substring(0, expression.length - 2);
+            fmt = MI2.FORMAT_SPEC_MAP[fmt];
         }
         expression = expression.replace(/"/g, '\\"');
 
         const createResp = await this.sendCommand(`var-create ${name} ${scope} "${expression}"`);
         let overrideVal = null;
         if (fmt && name !== '-') {
-            const formatResp = await this.sendCommand(`var-set-format ${name} ${MI2.FORMAT_SPEC_MAP[fmt]}`);
+            const formatResp = await this.sendCommand(`var-set-format ${name} ${fmt}`);
             overrideVal = formatResp.result('value');
         }
```

Comment:
By default fmt is null, it won't effect the performance. If user wants to change the format then it will be updated and  condition will be executed.